### ISSUE TITLE
Support ocp 4.4.7

### ISSUE THREAD

**Issue by tibcoplord** [openshift/windows-machine-config-bootstrapper#209]
With ocp 4.4.7  I see :

```
$ oc version -o json | jq -r '.serverVersion.gitVersion'
v1.17.1+f5fb168
```
This breaks tools/ansible/tasks/wsu/main.yaml ... I think we need to remove the +f5fb168 suffix

**Comment by aravindhp** [comment]
#202 has the fix backported. Please try again with that fix and let us know if you are still seeing issues.

**Comment by tibcoplord** [comment]
Well, I'm up-to-date on branch release-4.4 and its failing for me.  Temp fix was -

```
--- a/tools/ansible/tasks/wsu/main.yaml
+++ b/tools/ansible/tasks/wsu/main.yaml
@@ -22,7 +22,7 @@
 
     # Expected value of kubernetes_version in the form of 'v1.17.2'
     - name: Get kubernetes version
-      shell: "oc version -o json | jq -r '.serverVersion.gitVersion'"
+      shell: "oc version -o json | jq -r '.serverVersion.gitVersion' | cut -f1 -d+"
```

**Comment by aravindhp** [comment]
We ran into the same [issue](https://bugzilla.redhat.com/show_bug.cgi?id=1841140) fixed it in [this](https://github.com/openshift/windows-machine-config-bootstrapper/commit/e564a0d4ca84b4ef0f4c29e3d36762ac4c8ea63d) manner and confirmed that it works with 4.4. I am not sure why you need the extra fix. Please post the Ansible output and we will take a look.

cc: @mansikulkarni96

**Comment by tibcoplord** [comment]
This is what I see :

```
$ ansible-playbook -i hosts windows-machine-config-bootstrapper/tools/ansible/tasks/wsu/main.yaml -v
No config file found; using defaults

PLAY [localhost] ***********************************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************************
ok: [localhost]

TASK [Check if cluster is using ovn-kubernetes] ****************************************************************************************************************
changed: [localhost] => {"changed": true, "cmd": "oc get network cluster -o jsonpath='{.spec.networkType}'", "delta": "0:00:01.110579", "end": "2020-06-06 09:25:35.332306", "failed_when_result": false, "rc": 0, "start": "2020-06-06 09:25:34.221727", "stderr": "", "stderr_lines": [], "stdout": "OVNKubernetes", "stdout_lines": ["OVNKubernetes"]}

TASK [Check hybrid overlay is enabled] *************************************************************************************************************************
changed: [localhost] => {"changed": true, "cmd": "oc get network.operator cluster -o jsonpath='{.spec.defaultNetwork.ovnKubernetesConfig.hybridOverlayConfig.hybridClusterNetwork}'", "delta": "0:00:00.475724", "end": "2020-06-06 09:25:36.054976", "rc": 0, "start": "2020-06-06 09:25:35.579252", "stderr": "", "stderr_lines": [], "stdout": "[map[cidr:10.132.0.0/14 hostPrefix:23]]", "stdout_lines": ["[map[cidr:10.132.0.0/14 hostPrefix:23]]"]}

TASK [Fail if hybrid overlay is not enabled] *******************************************************************************************************************
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Get kubernetes version] **********************************************************************************************************************************
changed: [localhost] => {"changed": true, "cmd": "oc version -o json | jq -r '.serverVersion.gitVersion'", "delta": "0:00:00.536321", "end": "2020-06-06 09:25:36.869408", "failed_when_result": false, "rc": 0, "start": "2020-06-06 09:25:36.333087", "stderr": "", "stderr_lines": [], "stdout": "v1.17.1+f5fb168", "stdout_lines": ["v1.17.1+f5fb168"]}

TASK [Get kubernetes version without patch version] ************************************************************************************************************
changed: [localhost] => {"changed": true, "cmd": "version=v1.17.1+f5fb168 ; baseVersion=${version%-*} ; echo ${baseVersion%.*}  | tr -d v", "delta": "0:00:00.003475", "end": "2020-06-06 09:25:37.138836", "failed_when_result": false, "rc": 0, "start": "2020-06-06 09:25:37.135361", "stderr": "", "stderr_lines": [], "stdout": "1.17", "stdout_lines": ["1.17"]}

TASK [Get kubernetes changelog url] ****************************************************************************************************************************
changed: [localhost] => {"changed": true, "cmd": "echo \"https://github.com/kubernetes/kubernetes/blob/release-\"\\\n\"1.17/CHANGELOG/CHANGELOG-\"\\\n\"1.17.md\"\n", "delta": "0:00:00.002947", "end": "2020-06-06 09:25:37.407357", "rc": 0, "start": "2020-06-06 09:25:37.404410", "stderr": "", "stderr_lines": [], "stdout": "https://github.com/kubernetes/kubernetes/blob/release-1.17/CHANGELOG/CHANGELOG-1.17.md", "stdout_lines": ["https://github.com/kubernetes/kubernetes/blob/release-1.17/CHANGELOG/CHANGELOG-1.17.md"]}

TASK [Get kube node binary SHA] ********************************************************************************************************************************
[WARNING]: Consider using the get_url or uri module rather than running 'curl'.  If you need to use command because get_url or uri is insufficient you can add
'warn: false' to this command task or set 'command_warnings=False' in ansible.cfg to get rid of this message.

fatal: [localhost]: FAILED! => {"changed": true, "cmd": "curl -s \"https://github.com/kubernetes/kubernetes/blob/release-1.17/CHANGELOG/CHANGELOG-1.17.md\" | grep -A 1 \"v1.17.1+f5fb168\" | grep -A 1 node-windows | tail -n 1 | sed -e 's/.*<code>\\(.*\\)<\\/code>.*/\\1/'\n", "delta": "0:00:01.230737", "end": "2020-06-06 09:25:38.895839", "failed_when_result": true, "rc": 0, "start": "2020-06-06 09:25:37.665102", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

PLAY RECAP *****************************************************************************************************************************************************
localhost                  : ok=6    changed=5    unreachable=0    failed=1    skipped=1    rescued=0    ignored=0   
```

**Comment by tibcoplord** [comment]
I found that 4.4.4 works but 4.4.7 doesn't.

**Comment by aravindhp** [comment]
We are [unable to reproduce](https://bugzilla.redhat.com/show_bug.cgi?id=1841140#c5) this with 4.4.7. What host are you running Ansible from and what is your Ansible version?

**Comment by tibcoplord** [comment]
This is running on a linux host and with ansible :

```
ansible 2.9.0
  config file = None
  configured module search path = ['/Users/plord/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/plord/.local/lib/python3.6/site-packages/ansible
  executable location = /Users/plord/.local/bin/ansible
  python version = 3.6.8 (default, Apr  2 2020, 13:34:55) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

Do you see the same version (v1.17.1+f5fb168) as in my first post ?  This seems to be the root cause.

**Comment by tibcoplord** [comment]
Oh, I see the fix has been back ported to the release-4.4  ... I've just re-pulled and its turned up :-)
